### DISCUSSION TITLE
How to change default config of the `lazy.nvim`?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [NvChad/NvChad#2428] [Q&A] [answered]
I want to add `change_detection = { enabled = false }` to the `require("lazy.nvim").setup({...})`, but I haven't found an easy way, since only other plugins can be easily modified, but not the package manager plugin.

After some time, I was able to modify the lazy.nvim config like this:

```lua
-- chadrc.lua
M.lazy_nvim = (function()
  local default_config = require("plugins.configs.lazy_nvim")
  local custom_config = require("custom.configs.lazy_nvim")
  local config = vim.tbl_deep_extend("force", default_config, custom_config)
  return config
end)()
```

```lua
-- custom.configs.lazy_nvim.lua
return {
  change_detection = {
    enabled = false,
  },
}
```

Is this the optimal way? I didn't see a section about modifying lazy.nvim's config in the guide.

**Comment by siduck** [comment, marked as answer]
@Andrew15-5  just make M.lazy_nvim = { change_detection... blabla or require ("custom.configs.lazy_nvim")
